assignments/burmanm/week-3.scm

```diff
@@ -0,0 +1,279 @@
+;; Exercise 1.29
+
+;; TODO Somewhere a bug..
+
+;; n must be even
+(define h (/ (- b a) n))
+(define y (f (+ a (* k h))))
+(define (multiplier c n)
+  (cond ((= c n) 1)
+	((= c 1) 1)
+	((even? c) 4)
+	(else 2)))
+(define (integ f a b n)
+  (define (integ-iter f a b c n sum)
+    (if (> c n)
+	sum
+	(integ-iter f a b (+ c 1) n (+ sum (* (multiplier c n) (y c f a (h a b c)))))))
+  (integ-iter f a b 1 n 0))
+(define (integral f a b n)
+  (* (/ (h a b n) 3) (integ f a b n)))
+
+;; Exercise 1.30
+(define (inc n) (+ n 1))
+
+(define (sum term a next b)
+  (define (iter a result)
+    (if (> a b)
+	result
+	(iter (next a) (+ result (term a)))))
+  (iter a 0))
+
+;; Exercise 1.31
+
+(define (product term a next b)
+  (define (iter a result)
+    (if (> a b)
+	result
+	(iter (next a) (* result (term a)))))
+  (iter a 1))
+
+(define (product-recursive term a next b)
+  (if (> a b)
+      1
+      (+ (term a)
+	 (product-recursive term (next a) next b))))
+
+;; define factorial
+;; define approx-pi
+
+;; (define approx-pi 
+;; (product 
+
+;; Exercise 1.32
+
+;; (accumulate combiner null-value term a next b)
+(define (accumulate-recursive combiner null-value term a next b)
+  (if (> a b)
+      null-value
+      (combiner (term a)
+	 (accumulate-recursive combiner null-value term (next a) next b))))
+
+(define (accumulate combiner null-value term a next b)
+  (define (iter a result)
+    (if (> a b)
+	result
+	(iter (next a) (combiner result (term a)))))
+  (iter a null-value))
+
+(define (sum term a next b)
+  (accumulate + 0 term a next b))
+
+(define (product term a next b)
+  (accumulate * 1 term a next b))
+
+;; Exercise 1.33
+
+(define (filtered-accumulate filter combiner null-value term a next b)
+  (define (iter a result filter)
+    (if (> a b)
+	result
+	(iter (next a) (combiner (if (filter a) (term a) null-value)))))
+  (iter a null-value filter))
+
+(define (sum-square-prime a b)
+  (filtered-accumulate prime? + 0 square a inc b))
+
+(define (coprime? x) (= (gcd x n) 0))
```

Comment:
coprime should be when the gcd equals 1, and I think this needs to be part of the sum-coprime scope, otherwise n is unbound.